includes/gateway/class-omise-payment-atome.php

```diff
@@ -0,0 +1,196 @@
+                'title'       => __('Title', 'omise'),
+                'type'        => 'text',
+                'description' => __('This controls the title the user sees during checkout.', 'omise'),
+                'default'     => __('Atome', 'omise'),
+            ),
+
+            'description' => array(
+                'title'       => __('Description', 'omise'),
+                'type'        => 'textarea',
+                'description' => __('This controls the description the user sees during checkout.', 'omise')
+            ),
+        );
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function payment_fields()
+    {
+        parent::payment_fields();
+        $viewData = $this->validateMinRequiredAmount();
+
+        Omise_Util::render_view('templates/payment/form-atome.php', $viewData);
+    }
+
+    private function validateMinRequiredAmount()
+    {
+        $limits = [
+            'thb' => [
+                'min' => 20,
+                'max' => 150000,
+            ],
+            'sgd' => [
+                'min' => 1.5,
+                'max' => 20000,
+            ],
+            'myr' => [
+                'min' => 10,
+                'max' => 100000,
+            ]
+        ];
+
+        $currency = strtolower(get_woocommerce_currency());
+        $cartTotal = WC()->cart->total;
+
+        if (!isset($limits[$currency])) {
+            return [
+                'status' => false,
+                'message' => 'Currency not supported'
+            ];
+        }
+
+        $limit = $limits[$currency];
+
+        if ($cartTotal < $limit['min']) {
+            return [
+                'status' => false,
+                'message' => sprintf(
+                    "Amount must be greater than %u %s",
+                    number_format($limit['min'], 2),
+                    strtoupper($currency)
+                )
+            ];
+        }
+
+        if ($cartTotal > $limit['max']) {
+            return [
+                'status' => false,
+                'message' => __(
+                    'Amount must be less than %1 %2',
+                    number_format($limit['max'], 2),
+                    $currency
+                )
+            ];
+        }
+
+        return ['status' => true];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function charge($order_id, $order)
+    {
+        $currency = $order->get_currency();
+        $default_phone_selected = $_POST['omise_atome_phone_default'];
+        $phone_number = isset($default_phone_selected) && 1 == $default_phone_selected ?
+            $order->get_billing_phone()
+            : sanitize_text_field($_POST['omise_phone_number']);
+
+        return OmiseCharge::create([
+            'amount' => Omise_Money::to_subunit($order->get_total(), $currency),
+            'currency' => $currency,
+            'description' => apply_filters(
+                'omise_charge_params_description',
+                'WooCommerce Order id ' . $order_id,
```

Comment:
We define this exact string across all payment methods. Is there a way we can factor it out?

Reply:
We should. There are other parameters like `amount`, `currency` and `metadata` as well. Let't do it a separate ticket because we need to test all the payment methods after the change.